src/Handlers/HandlerGroup.php

```diff
@@ -46,4 +49,31 @@ public function getScopes(): array
     {
         return $this->scopes;
     }
+
+    public function getHash(): string
+    {
+        $data = [
+            'disabled' => $this->disabled,
+            'constraints' => $this->constraints,
+            'tags' => $this->tags,
+        ];
+
+        return (string)crc32(serialize($data));
+    }
+
+    public function throttle(int $maxAttempts, int $decaySeconds = 60, ?string $key = null): self
+    {
+        $this->rateLimit = new RateLimit(
+            maxAttempts: $maxAttempts,
+            decaySeconds: $decaySeconds,
+            key: $key ?? $this->getHash(),
```

Comment:
getHash can be "old" if you write

->middleware(x)
->throttle(x)
->middleware(x)